### PR TITLE
Set passwords

### PR DESCRIPTION
**Pull request by imobachgs**
Check that password fields in `/etc/shadow` are ok.

### CODE REVIEW

**Comment by schubi2** [comment]
I have been a little bit confused with the "||" in the checking script :-). But you are right . --> LGTM

**Comment by imobachgs** [comment]
@schubi2 I must admit that it can be confusing, so I've added a comment to make it clear.

Thanks!

File: aytests/set_passwords.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# Check if all users has some value in the password field
+# (bsc#973639, bsc#974220, bsc#971804 and bsc#965852)
+
+set -e -x
+
+# It's OK if we DO NOT find a blank password.
+getent shadow | grep -P "^[^:]+::" || echo "AUTOYAST OK"
```
Comment (mvidner) [inline aytests/set_passwords.sh]: `-P` for `--perl-regexp` seems too strong. `-E`, `--extended-regexp` works just as well.

**Comment by mvidner** [comment]
Shouldn't  set_passwords.sh be added to most `aytests/*.list` ? in particular, what about upgrade_sles11.list ?

**Comment by imobachgs** [comment]
@mvidner Ok, I'll change it.

On the other hand, adding this test to almost all *.list, makes me wonder if we shoud have support for "including" other lists (so we could just set `include sles12-common.list`).

**Comment by imobachgs** [comment]
Well, we can achieve it using the rspec helper:

``` ruby
include_examples "test_scripts", "sles12-common"
```

But as we're running these tests in openQA, we would need to provide an alternative mechanism.

**Comment by imobachgs** [comment]
I've added the `set_passwords.sh` tests to all lists. At the end of the current sprint I'll try to add a feature to include list of tests.

**Comment by mvidner** [comment]
Thanks!
LGTM

**Comment by imobachgs** [comment]
Thanks!

**Pull request merged**